backend/app/users/sensitive_personal_data/test_repository.py

```diff
@@ -0,0 +1,208 @@
+        aes_encrypted_data=get_random_printable_string(100),
+        created_at=datetime.now()
+    )
+
+
+class TestFindById:
+    @pytest.mark.asyncio
+    async def test_success(self, mocked_users_database):
+        users_db = await mocked_users_database
+
+        # GIVEN sensitive personal data
+        given_sensitive_data = _get_new_sensitive_personal_data()
+
+        # AND the data is saved in the database
+        await users_db.get_collection(Collections.SENSITIVE_PERSONAL_DATA).insert_one(given_sensitive_data.model_dump())
+
+        # WHEN we try to find the sensitive data by user_id
+        repository = SensitivePersonalDataRepository(users_db)
+        actual_sensitive_data = await repository.find_by_user_id(given_sensitive_data.user_id)
+
+        # THEN the sensitive personal data is found
+        # AND it is equal to the same sensitive data in the database
+        assert actual_sensitive_data == given_sensitive_data
+
+    @pytest.mark.asyncio
+    async def test_find_by_id_not_found(self, mocked_users_database):
+        # set up
+        users_db = await mocked_users_database
+        repository = SensitivePersonalDataRepository(users_db)
+
+        # GIVEN some random user id
+        given_user_id = get_random_printable_string(10)
+
+        # WHEN we try to find sensitive data by user_id
+        actual_sensitive_data = await repository.find_by_user_id(given_user_id)
+
+        # THEN the sensitive personal data is not found
+        assert actual_sensitive_data is None
+
+    @pytest.mark.asyncio
+    async def test_find_by_id_throws_an_error(self, mocked_users_database, mocker):
+        users_db = await mocked_users_database
+        repository = SensitivePersonalDataRepository(users_db)
+
+        # GIVEN some random user id
+        given_user_id = get_random_printable_string(10)
+
+        # AND some random error
+        given_error = Exception('given error message')
+
+        # AND repository find one throws an error
+        repository_find_by_id = mocker.spy(repository._collection, 'find_one')
+        repository_find_by_id.side_effect = given_error
+
+        # WHEN we try to find sensitive data by user_id
+        # THEN an error should be thrown
+        with pytest.raises(Exception) as error_info:
+            await repository.find_by_user_id(given_user_id)
+
+        # AND error message should be the same as given
+        assert error_info.value == given_error
+
+
+class TestCreate:
+    @pytest.mark.asyncio
+    async def test_success(self, mocked_users_database):
+        users_db = await mocked_users_database
+        repository = SensitivePersonalDataRepository(users_db)
+
+        # GIVEN some sensitive personal data
+        given_sensitive_data = _get_new_sensitive_personal_data()
+
+        # WHEN we create the sensitive personal data
+        actual_inserted_id = await repository.create(given_sensitive_data)
+
+        # THEN the inserted_id is not None and it is a string
```

Comment:
small thing... this is less of an expectation and more of a guard, if we dont use THEN here we can have the test read easier... right?

Reply:
like this
```python
# GIVEN some sensitive personal data
# ... 
# WHEN we create the sensitive personal data
# ...
# guard: the inserted_id is not None and it is a string
# ...
# AND the sensitive personal data is saved in the database
# ...
# THEN expect the saved sensitive personal data to be equal to the given data
```

Reply:
what do you think? does that make sense? I think it would be breaking the pattern we use so far, but worth looking at?

Reply:
it is not a guard it is the expectation of the function.
it is the return value ,we expect it to return "the objectid of the newly created as a string"